dimscord/helpers/channel.nim

```diff
@@ -0,0 +1,124 @@
+template pin*(m: Message, reason = ""): Future[void] =
```

Comment:
Well that's because you cannot have "async templates", the trick is to have`Future[T]` as the return type and use `await` on it. It works asynchronously because the template just low-key replace itself with an async proc so you do have to use `await` , just not the `async` pragma.